docs/modules/ROOT/pages/explanations/decisions/release_process.adoc

```diff
@@ -0,0 +1,125 @@
+For instance, the crossplane helm chart had to be patched with a comp-function sidecar pod.
+
+== Proposals
+
+Proposal 1::
+Keep package release.
+
+We may try to keep the current package approach with minimal changes in order to fix the problem described above.
+The change implies getting `appcat` package to a new GitHub repository with its own release process.
+This approach will eliminate the problem of master branch being rolled out after a tag.
+The same automation we currently use for package release can be applied to this solution.
+
+Advantages::
+
+* Minimal changes to current implementation;
+* No need to adapt to a new release process.
+
+Disadvantages::
+
+* The package feature is not intended to be used for our current needs;
+* Adds another unnecessary layer of abstraction in our release process;
+* Overly complicated;
+* `commodore-defaults` must be changed from component to package release.
+
+Proposal 2::
+Revert to component release.
+
+This proposal implies removing package feature from `component-appcat` repository
+
+Advantages::
+
+* Easy and straightforward release process;
+* `commodore-defaults` repository becomes relevant to the release process.
+
+Disadvantages::
+
+* We change again the release process paradigm;
+* We need to add the remaining dependencies, such as component crossplane into `component-appcat`.
+
+=== Pipeline Enhancement Proposal
+
+To automate further our release process we can add specific GitHub Actions to each application and/or `component-appcat` repositories.
+
+Enhancement 1::
+Application Release from their repository
+
+. New PR in any application managed by `component-appcat` would trigger:
+* A new PR on `component-appcat` side with the same branch name
+* Create all container images related to this branch
+* Any other operations relevant to the release process that are missing here
+. (Optional) Manual GitHub release action for PR branch in lab cluster from `component-appcat`.
+. Update any branches from both repos, application and/or component after a review or bug.
+. Once both PRs are ready and approved, label the application PR with one of bug/minor/major values.
+. Merge the application PR into master branch and that would trigger:
+* A release process for application based on the label (ex:. label bug - from v2.3.1 to v2.3.2)
+* Create a new commit into `component-appcat` PR branch with the latest application dependencies and golden tests on top of any previous commits.
+* Delete any obsolete branch or any other relevant resources.
+. `component-appcat` PR ready for review:
+* If no engineer changes have been made to the component PR then review it yourself and merge
+* If engineer changes have been made to the component PR then wait for others to review.
+. (Optional) Once `component-appcat` is approved,label the PR with one of bug/minor/major values and merge it and that would trigger:
```

Comment:
Maybe the `component-appcat` PR could by default get the same labels as the application PR? Of course, it can then be changed manually if needed.

Reply:
Yeah that's another small automation that we can add but what if you have a small change in appcat and a full blown feature in component? You would still need to tackle the labels. Not sure if it's worth to automate this, it can backfire.